### PR TITLE
refactor and add some checker

### PR DESCRIPTION
**Pull request by siddontang**

### CODE REVIEW

File: compare.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
+// Copyright 2015 Reborndb Org. All Rights Reserved.
```
Comment (qiuyesuifeng) [inline compare.go]: remove head comment

**Comment by qiuyesuifeng** [comment]
need godropbox license, see https://github.com/dropbox/godropbox/blob/master/license.txt.

**Comment by coocood** [comment]
LGTM

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**